### PR TITLE
DG and other UI updates

### PR DESCRIPTION
**Pull request by NicholasTanYY**
- Added and removed functions to measure terminal size
- and subsequently centralise avatar and week/month view
- and choose avatar size based on terminal size

because there are many issues with implementation later on, and many forseeable bugs.

I will fix the rendering of the avatar in the next iteration.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -7,6 +7,72 @@
...
+
+### Overview: <br> 
+The UiRenderer component is responsible for rendering the user interface. It is used to display messages, week views as well as month views to the user.
+
+### How it works:
+1. When week view is requested, the UiRenderer component will render the week view.
+2. This is done by printing the table in ASCII art format, with the days of the week as columns.
+3. The week view will display the tasks for the week, with the tasks for each day displayed in the respective columns.
+4. The UiRenderer component will also display the current date at the top of the week view.
+5. Similarly, when month view is requested, the UiRenderer component will render the month view.
```
Comment (Aak242) [inline docs/DeveloperGuide.md:21]: Clear explanation of the UIRenderer class and how it works to show the week and month views!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -7,6 +7,72 @@
...
+
+#### Method signature:
+```
+public static void printWeekHeader(LocalDate startOfView, DateTimeFormatter dateFormatter, boolean isMonthView)
+```
+
+#### Parameters
+- startOfView: The starting date of the week view.
+- dateFormatter: Formatter for displaying dates.
+- isMonthView: A boolean flag indicating if the month view is being rendered.
```
Comment (Aak242) [inline docs/DeveloperGuide.md:36]: As mentioned by Clarence on my PR, we should discuss whether we should include details like the parameters of the method

File: src/main/java/ui/AvatarUi.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            "▓▓▓▓████▓█▓▓▓▓▓▓▓▓▓▓█▓▓▓▓▓▓▓▓██▓░░░░░██████░░░░░▓██▓▓▓▓▓▓▓▓█▓▓▓▓▓▓▓▓▓▓▓██████▓██\n" +
+            "████████▓██▓▓▓▓▓▓▓▓▓██▓▓▓▓▓▓▓██▓▒░░░▒██████▒░░░▒▓██▓▓▓▓▓▓▓▓█▓▓▓▓▓▓▓▓▓██▓█████▓▓▓\n" +
+            "▓▓▓████████████▓▓▓▓▓▓██▓▓▓▓▓▓██▓▓░░░▓██████▒░░░▓▓██▓▓▓▓▓▓███▓▓▓▓▓▓▓███████████▓▓\n" +
+            "░░░█████████████████▓██████████▓▓▓░░▓██████▓░░▓▓▓███████████████████████████████\n" +
+            "▒▒▓█████████████████████████████▓▓▒░▓███████░▒█▓▓█████████████████████████████▓▓\n" +
+            "█▓█████████████████████████████████▒▓███████░▓▓▓████████████▓▓▓▓▓█████████████▒▒\n" +
+            "█▒██████████████████████████████████████████▓████████████▓▓▓▓█████████████████▓▒\n" +
+            "█▒████████████████████████████████████████████████████████████████████████████▓▒\n" +
+            "▓▓█████████████████████████████████████████████████████████████████████████████▒\n" +
+            "▓▓█████████████████████████████████████████████████████████████████████████████▒\n";
```
Comment (Aak242) [inline src/main/java/ui/AvatarUi.java:45]: Thanks for adding in such a detailed an professional looking ASCII avatar!

File: src/main/java/ui/UiRenderer.java [hunk 4]
```diff
@@ -16,12 +16,12 @@ public class UiRenderer {
         "Wednesday", "Thursday", "Friday", "Saturday"};
 
     private static final int SPACE_COUNT = 10;
-    private static final String SINGLE_HORIZONTAL_DIVIDER = "+" + "-".repeat(SPACE_COUNT + 2);
+    private static final String SINGLE_HORIZONTAL_DIVIDER = "+" + "-".repeat(SPACE_COUNT);
```
Comment (Aak242) [inline src/main/java/ui/UiRenderer.java:19]: Good use of the repeat function to automatically add spaces and store them within this final variable

**Review by Aak242 — COMMENTED** [review]
Overall, the UI related elements have been streamlined well and the explanation of the same in the developer guide is quite detailed. LGTM!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -7,6 +7,72 @@
 ical4J Library: https://www.ical4j.org/
 
 ## Design & implementation
+## UiRenderer Component
+### API: [UiRenderer.java](https://github.com/AY2324S2-CS2113-W13-2/tp/blob/master/src/main/java/ui/UiRenderer.java)
+
+### Overview: <br>
```
Comment (clarencepohh) [inline docs/DeveloperGuide.md:13]: Is the <br> necessary? I think the header already creates a line break.

File: src/main/java/ui/AvatarUi.java (hunk 3 above)
Comment (clarencepohh) [reply src/main/java/ui/AvatarUi.java:45]: Looks amazing!

**Review by clarencepohh — COMMENTED** [review]
Looks good!

**Pull request merged**